test/pvtk_files.jl

```diff
@@ -0,0 +1,187 @@
+# test parallel VTK files
+using WriteVTK, Test, ReadVTK
+
+# (1) Generate parallel structured input files
+
+# Global grid
+xs_global = range(0, 2; length = 15)
+ys_global = range(-1, 1; length = 12)
+zs_global = range(0, 1; length = 4)
+v_global =  range(0,1,length=3)
+
+extents = [
+    ( 1:10,  1:5, 1:4),  # process 1
+    (10:15,  1:5, 1:4),  # process 2
+    ( 1:10, 5:12, 1:4),  # process 3
+    (10:15, 5:12, 1:4),  # process 4
+]
+
+saved_files = Vector{Vector{String}}(undef, 4)  # files saved by each "process"
+
+# Write *.pvti file
+for part = 1:4
+    is, js, ks = extents[part]  # local indices
+    xs, ys, zs = xs_global[is], ys_global[js], zs_global[ks]  # local grid
+    xs_c, ys_c, zs_c = xs[1:end-1], ys[1:end-1], zs[1:end-1]
+
+    saved_files[part] = pvtk_grid(
+            "fields", xs, ys, zs;
+            part = part, extents = extents,
+        ) do pvtk
+        pvtk["Temperature"] = [x + 2y + 3z for x ∈ xs, y ∈ ys, z ∈ zs]
+        pvtk["Velocity"] = [x + 2y + 3z + v for v ∈ v_global, x ∈ xs, y ∈ ys, z ∈ zs]
+        pvtk["Pressure"] = [x + 2y + 3z for x ∈ xs_c, y ∈ ys_c, z ∈ zs_c]
+        pvtk["Phase"] = [trunc(Int64, x*15) for x ∈ xs_c, y ∈ ys_c, z ∈ zs_c]
+    end
+end
+
+# write *.pvtr file
+for part = 1:4
+  is, js, ks = extents[part]  # local indices
+  xs, ys, zs = xs_global[is], ys_global[js], zs_global[ks]  # local grid
+  xs_c, ys_c, zs_c = xs[1:end-1], ys[1:end-1], zs[1:end-1]
+  saved_files[part] = pvtk_grid(
+          "fields", Vector(xs), Vector(ys), Vector(zs);
+          part = part, extents = extents,
+      ) do pvtk
```

Comment:
I attach this here but it really applies to multiple locations:

I strongly recommend to remove all files that are generated by your tests in the beginning, and then again at the end. This way, you minimize the chance of accidentally ending up with an old test file that was generated during a previous run and which could compromise the validity of subsequent test runs.

For this purpose, we create a test file directory in
https://github.com/trixi-framework/ReadVTK.jl/blob/3abe6b8b87bf1e79ec8f34de2158bc20fa5331f2/test/runtests.jl#L19-L21
and delete it at the end again
https://github.com/trixi-framework/ReadVTK.jl/blob/3abe6b8b87bf1e79ec8f34de2158bc20fa5331f2/test/runtests.jl#L373.

If you use a path to the test directory here and in the other locations, you will automatically have everything removed in the end again:
```suggestion
  path = joinpath(TEST_EXAMPLES_DIR, "fields")
  saved_files[part] = pvtk_grid(
          path, Vector(xs), Vector(ys), Vector(zs);
          part = part, extents = extents,
      ) do pvtk
```
See also e.g. here:
https://github.com/trixi-framework/ReadVTK.jl/blob/3abe6b8b87bf1e79ec8f34de2158bc20fa5331f2/test/runtests.jl#L347-L350

Reply:
yes, very good point - thanks!